sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientTests.cs

```diff
@@ -0,0 +1,151 @@
+            Assert.AreEqual((int)HttpStatusCode.Accepted, response.GetRawResponse().Status);
+            Assert.AreEqual("cad9df7b-f3ac-4c53-96f7-c76e7437b3c1", response.Value.CallConnectionId);
+        }
+
+        [TestCaseSource(nameof(TestData_JoinCall))]
+        public void JoinCall_Returns202Accepted(string serverCallId, CommunicationIdentifier source, JoinCallOptions joinCallOptions)
+        {
+            CallingServerClient callingServerClient = CreateMockCallingServerClient(202, CreateOrJoinCallPayload);
+
+            var response = callingServerClient.JoinCall(serverCallId, source, joinCallOptions);
+            Assert.AreEqual((int)HttpStatusCode.Accepted, response.GetRawResponse().Status);
+            Assert.AreEqual("cad9df7b-f3ac-4c53-96f7-c76e7437b3c1", response.Value.CallConnectionId);
+        }
+
+        [TestCaseSource(nameof(TestData_JoinCall))]
+        public void JoinCallAsync_Returns404NotFound(string serverCallId, CommunicationIdentifier source, JoinCallOptions joinCallOptions)
+        {
+            CallingServerClient callingServerClient = CreateMockCallingServerClient(404);
+
+            RequestFailedException? ex = Assert.ThrowsAsync<RequestFailedException>(async () => await callingServerClient.JoinCallAsync(serverCallId, source, joinCallOptions).ConfigureAwait(false));
+            Assert.NotNull(ex);
+            Assert.AreEqual(ex?.Status, 404);
+        }
+
+        [TestCaseSource(nameof(TestData_JoinCall))]
+        public void JoinCall_Returns404NotFound(string serverCallId, CommunicationIdentifier source, JoinCallOptions joinCallOptions)
+        {
+            CallingServerClient callingServerClient = CreateMockCallingServerClient(404);
+
+            RequestFailedException? ex = Assert.Throws<RequestFailedException>(() => callingServerClient.JoinCall(serverCallId, source, joinCallOptions));
+            Assert.NotNull(ex);
+            Assert.AreEqual(ex?.Status, 404);
+        }
+
+        private static IEnumerable<object?[]> TestData_CreateCall()
+        {
+            return new[]
+            {
+                new object?[]
+                {
+                    new CommunicationUserIdentifier("8:acs:resource_source"),
+                    new CommunicationIdentifier[]
+                    {
+                        new CommunicationUserIdentifier("8:acs:resource_target"),
+                        new PhoneNumberIdentifier("+14250001234")
+                    },
+                    new CreateCallOptions(
+                        new Uri("https://bot.contoso.com/callback"),
+                        new[]
+                        {
+                            MediaType.Video
+                        },
+                        new[]
+                        {
+                            EventSubscriptionType.ParticipantsUpdated
+                        }
+                    )
+                    {
+                        AlternateCallerId = new PhoneNumberIdentifier("+17781234567"),
```

Comment:
We usually try to use `+1425XXXXXX` as a fake number. `+17781234567` could actually be a real number

Reply:
Changed to "+1425XXXXXXX".

Reply:
Hey Zihan, sorry I wasn't clear enough, I meant for `+1425XXXXXX` as a format, the X's should actually be numbers. The actual number we use as fake numbers is usually `+14255550123`

Reply:
😃Addressed this issue. change it to +14255550123